pkg/pillar/base/logobjecttypes.go

```diff
@@ -205,7 +205,7 @@ type LoggableObject interface {
 // This is critical to keep e.g., the subscribers and publishers for the same
 // objects apart when those subscribers and publishers run in the same process
 func (object *LogObject) mapKey(key string) string {
-	return fmt.Sprintf("%s:%p", key, object)
+	return fmt.Sprintf("%s:%p", key, object.logger)
```

Comment:
The key question here is whether we always create unique loggers for each microservice running in the same process.
@gkodali-zededa do we also need to keep them apart when we have a microservice subscribe to something it publishes? In that case using the logger wouldn't be sufficient.

Reply:
As I can see we create separate logger for each service https://github.com/lf-edge/eve/blob/b2471c507920e3c5888ca9740c1dc271d5bf917c/pkg/pillar/zedbox/zedbox.go#L221

Reply:
In the case we have a publisher and subscriber in the same service (which we have at least in nim and probably elsewhere) they will not be treated as separate in this log map.

Also, you don't even consider the type of the object. We have the same UUID (or UUID+version, or sha) as key for objects of completely different types.

Reply:
Oops this is using the LogKey and not the Key and the LogKey() functions all use obe of the define LogKey strings which capture the type of the device.

But we still need to distinguish between a publish and subscribe of the identical thing. One way to do that would be to pass another arg from pubsub to LogCreate(), LogModify() and LogDelete() to indicate "publish" vs, "subscribe" and include that in they key used for this map.

Reply:
Do you mean that we have publisher and subscriber of the same type in the same agent? Let me check.

Reply:
I think care is required because in the LogModify we typically fatal if the map entry is not found. So we can't accidentally delete from the map.

Maybe it makes sense to split out that commit so we can merge the other commits?

Reply:
Yes, sounds reasonable.